File: python/ABC/ABC250A.py
H, W = map(int, input().split())
R, C = map(int, input().split())

ans = 0
if R > 1:  # 上
    ans += 1
if C > 1:  # 左
    ans += 1
if R < H:  # 下
    ans += 1
if C < W:  # 右
    ans += 1
print(ans)

File: C/ABC/ABC250A.c
#include <stdio.h>

int main()
{
    int H, W;
    int R, C;
    scanf("%d %d", &H, &W);
    scanf("%d %d", &R, &C);

    int ans = 0;
    if (R > 1) // 上
        ans++;
    if (C > 1) // 左
        ans++;
    if (R < H) // 下
        ans++;
    if (C < W) // 右
        ans++;

    printf("%d\n", ans);
    return 0;
}
